src/Npgsql/NpgsqlDataReader.cs

```diff
@@ -1466,7 +1471,7 @@ public override object GetProviderSpecificValue(int ordinal)
             }
             catch (NpgsqlSafeReadException e)
             {
-                throw e.InnerException;
+                throw e.InnerException!;
```

Comment:
I don't think this should be necessary - NpgsqlSafeReadException should always wrap an inner exception. We can add a `new` InnerException property to indicate that.

Reply:
It make sense to remove this type because it hides the original stack trace, so leaving as is.

Reply:
This type signals that the connection is still OK and wasn't broken, and the original stack trace is still available in the inner exception.

In any case, if we're not removing it, then why not remove the bang by introducing a null-nullable InnerException as I proposed above?